src/Analyser/MutatingScope.php

```diff
@@ -2730,6 +2730,11 @@ public function enterClosureCall(Type $thisType, Type $nativeThisType): self
 		$nativeExpressionTypes = $this->nativeExpressionTypes;
 		$nativeExpressionTypes['$this'] = ExpressionTypeHolder::createYes(new Variable('this'), $nativeThisType);
 
+		$thisObjectClassNames = $thisType->getObjectClassNames();
+		if (count($thisObjectClassNames) > 1) {
+			throw new ShouldNotHappenException();
```

Comment:
Can be more than one. You can typically have intersection type of ThisType and another type, for example:

```php
if ($this instanceof FooInterface) {
    // voila
}
```

You can change MutatingScope internals to correctly represent this.

Reply:
Should be tested of course :)

Reply:
I pushed a fixup trying to adapt it. I'm not fully sure about it though. I added a test that at least leads now to passing 2 classNames there, which would have thrown an exception before, but the assertion was the same before. In theory something should be smarter now around `canAccessClassMember()` but I couldn't figure out how to test this yet :/